tests/Credentials/InstanceProfileProviderTest.php

```diff
@@ -200,4 +202,63 @@ public function testThrowsExceptionOnInvalidJsonResponse()
             ['retries' => 0]
         )->wait();
     }
+
+    /**
+     * @expectedException \Aws\Exception\CredentialsException
+     * @expectedExceptionMessage Networking error
+     */
+    public function testThrowsExceptionOnNetorkRetryExhaustion()
+    {
+        $client = function () {
+            return Promise\rejection_for([
+                'exception' => new RequestException(
+                    'message',
+                    new Request('GET', 'http://example.com'),
+                    new Response('500')
+                )
+            ]);
+        };
+
+        $args = [
+            'profile' => 'foo',
+            'client' => $client,
+            'retries' => 0
+        ];
+        $provider = new InstanceProfileProvider($args);
+        $c = $provider()->wait();
+    }
+
+    public function testNetworkingErrorsAreRetried()
+    {
+        $retries = 1;
+
+        $t = time() + 1000;
+        $result = json_encode($this->getCredentialArray('foo', 'baz', null, "@{$t}"));
+        $responses = [new Response(200, [], Psr7\stream_for($result))];
+
+        $client = function () use (&$retries, $responses) {
+            if (0 === --$retries) {
```

Comment:
With the `--` before `$retries`, won't this only ever execute once without ever hitting the rejection below?

Reply:
Yes, should be a post-decrement so the rejection is thrown on the first request.